qt/python/mantidqt/mantidqt/widgets/rawdataexplorer/view.py

```diff
@@ -0,0 +1,328 @@
+            message = "Could not open view:\n{0}".format(e)
+            logger.error(message)
+            message_box = QMessageBox()
+            message_box.setText(message)
+            message_box.exec()
+            self.on_close()
+            return
+
+        if self._type == self.IVIEW:
+            self._widget.show_view()
+            self._widget.container.closing.connect(self.on_close)
+            self.sig_request_close.connect(self._widget.container.emit_close)
+            self.sig_request_close.connect(lambda: self.on_close(None, False))
+
+        if self._type == self.SVIEW:
+            self._widget.view.show()
+            self._widget.view.close_signal.connect(self.on_close)
+            self.sig_request_close.connect(self._widget.view.emit_close)
+            self.sig_request_close.connect(lambda: self.on_close(None, False))
+
+        if self._type == self.PLOT2D:
+            self._widget.canvas.mpl_connect("close_event", self.on_close)
+            self.sig_request_close.connect(lambda: plt.close(self._widget))
+            self.sig_request_close.connect(lambda: self.on_close(None, False))
+
+        if self._type == self.PLOT1D:
+            self._widget.canvas.mpl_connect("close_event", self.on_close)
+            self.sig_request_close.connect(lambda: plt.close(self._widget))
+            self.sig_request_close.connect(lambda: self.on_close(None, False))
+
+        if self._type == self.PLOTSPECTRUM:
+            self._widget.canvas.mpl_connect("close_event", self.on_close)
+            self.sig_request_close.connect(lambda: plt.close(self._widget))
+            self.sig_request_close.connect(lambda: self.on_close(None, False))
+
+        self._presenter.get_main_view().fileTree.set_ignore_next_focus_out(True)
+
+    def get_widget(self, workspace_name: str):
+        """
+        Create the appropriate widget to show.
+        @param workspace_name: name of the workspace
+        """
+        if self._type == self.IVIEW:
+            parent, flags = get_window_config()
+            self._widget = get_instrumentview(workspace_name, True, parent, flags, use_thread=False)
+        elif self._type == self.SVIEW:
+            self._widget = SliceViewer(ws=mtd[workspace_name])
+        elif self._type == self.PLOT1D:
+            self._widget = plotBin(workspace_name, 0, error_bars=True)
+        elif self._type == self.PLOT2D:
+            self._widget = pcolormesh([workspace_name])
+        elif self._type == self.PLOTSPECTRUM:
+            self._widget = plotSpectrum(workspace_name, 0, error_bars=True)
+
+    def change_workspace(self, workspace_name: str):
+        """
+        Change the workspace displayed by the current widget.
+        @param workspace_name: the name of the new workspace
+        """
+        if self._type == self.IVIEW:
+            self._widget.replace_workspace(workspace_name)
+        elif self._type == self.SVIEW:
+            # TODO find how to replace slice viewer's current workspace
```

Comment:
Is the slice view's current workspace not replaced at all then? Could we improve on that?

Reply:
It certainly looks like that, though without test data it is a bit hard to be sure of the actual behavior. In any case putting _something_ in this clause would be a good first step

Reply:
Ok, after a look at the slice viewer code:
There is no method for replacing the workspace currently shown BUT if the workspace shown is replaced in the ADS while keeping the same name, the slice viewer adapts.
That can be an ugly hack to make this work if the need arises in the future (for now nothing actually needs that).

Reply:
Since nothing currently uses that, it could be either removed altogether or a warning raised when a user tries to visualise something in this mode. In the latter case, also a comment could be added here in the code.